odoo/addons/base/models/ir_module.py

```diff
@@ -999,6 +999,27 @@ def _compute_state(self):
         for dependency in self:
             dependency.state = dependency.depend_id.state or 'unknown'
 
+    @api.model
+    def all_dependencies(self, module_names):
+        to_search = {key: True for key in module_names}
+        res = {}
+        def search_direct_deps(to_search, res):
+            to_search_list = list(to_search.keys())
+            dependencies = self.web_search_read(domain=[("module_id.name", "in", to_search_list)], specification={"module_id":{"fields":{"name":{}}}, "name": {}, })["records"]
```

Comment:
`web_search_read` should not be used in the addon `base`. Anyway, this method should be moved to `web` addon